src/APIBasedAuth.ts

```diff
@@ -1,57 +1,55 @@
-import axios, { AxiosError, AxiosInstance, AxiosResponse } from 'axios';
-
-// exporting just for jest test coverage
-export const DEFAULT_BASE_URL = 'https://apps.us.lifeomic.com/auth/v1/api';
-
-// exporting just for jest test coverage
-export const formatAxiosError = <ErrorResponse = any>(
-  error: AxiosError<ErrorResponse> | unknown
-) => {
-  if (!axios.isAxiosError(error)) {
-    throw new AxiosError(String(error));
-  }
-  throw error.response?.data;
-};
+import axios, { AxiosInstance, AxiosResponse } from 'axios';
+import { window } from './globals';
+import { DEFAULT_BASE_URL, formatAxiosError } from './utils';
 
 class APIBasedAuth {
   private client: AxiosInstance;
-  private clientId: APIBasedAuth.Config['clientId'];
+  private clientOptions: APIBasedAuth.Config;
   private session?: APIBasedAuth.Session;
-
-  constructor({ clientId, ...options }: APIBasedAuth.Config) {
+  private tokens?: APIBasedAuth.Tokens;
+
+  constructor({
+    baseURL,
+    clientId,
+    storage,
+    storageKeys = {}
+  }: APIBasedAuth.Config) {
     if (!clientId) {
       throw new Error('APIBasedAuth param clientId is required');
     }
 
-    this.client = axios.create({
-      ...options,
-      baseURL: options.baseURL || DEFAULT_BASE_URL
-    });
+    this.client = axios.create({ baseURL: baseURL || DEFAULT_BASE_URL });
 
-    this.clientId = clientId;
+    this.clientOptions = {
+      clientId,
+      storage: storage || window.localStorage,
```

Comment:
safe? not entirely cause the potential exists where this is being used in a react-native context and no storage is provide and in that case, window.localStorage isn't a thing. so I could not use storage if nothing is provided. otherwise we'd have to add some sort of platform selector in which adds the react-native lib or requires a type on initialization to let us know what platform we're dealing with and if we can use window.localStorage as a default fallback

Reply:
The existing `LOAuth` client already:
- defines a highly generalized type for a "Storage" abstraction,
- accepts it in the constructor, and
- has a pattern for using it internally.

Any reason not to just re-use the types + patterns defined there?

I think it's better to keep any platform-specific code out of the class, and just require providing a `storage` parameter.

Reply:
If it helps, the reason that each token had its own key is that on mobile we were hitting a size limit warning that might be treated as throwing an error in a future version of expos secure store https://docs.expo.dev/versions/latest/sdk/securestore/

Reply:
@smolster @aecorredor so theoretically we need to keep that known issue for expo in mind for the solution we make. so having a generic single function for setting storage which won't allow us to json stringify the tokens object and store it. knowing this, thoughts on the solution I have with individual key-value setting for everything?

Reply:
@virdesai I forgot about the scenario of wanting to store the `session` + token in different places.